File: 21_03_2022/Alpha_garland.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

string reversed(string n){
    reverse(n.begin(),n.end());
    return n;
}

int main(){
    
    string n;
    cin>>n;

    string s = "";
    for(int i=0;i<n.length();i++){
        if(isalpha(n[i])){
            s += n[i];
        }
        else{
            bool rev = true;
            string x = "";
            for(int j=0;j<n[i]-'0';j++){
                if(rev){
                    x += reversed(s);
                    rev = false;
                }   
                else{
                    x += s;
                    rev = true;
                }
            }
            s += x;
        }
    }

    cout<<s<<endl;

    return 0;
}

File: 21_03_2022/Alpha_garland.py
n = input()
s = ""
for i in n:
    if(i.isalpha()):
        s += i
    else:
        reverse = True
        x = ""
        for j in range(int(i)):
            if(reverse):
                x += s[::-1]
                reverse = False
            else:
                x += s
                reverse = True
        s += x
print(s)